drivers/interrupt_controller/intc_exti_stm32_priv.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2025 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_INTERRUPT_CONTROLLER_INTC_EXTI_STM32_PRIV_H_
+#define ZEPHYR_DRIVERS_INTERRUPT_CONTROLLER_INTC_EXTI_STM32_PRIV_H_
+
+#include <zephyr/device.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/drivers/interrupt_controller/intc_exti_stm32.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32g0_exti)
+
+#define EXTI_IS_ACTIVE_RISING_FLAG_0_31(line)  LL_EXTI_IsActiveRisingFlag_0_31(line)
+#define EXTI_IS_ACTIVE_FALLING_FLAG_0_31(line) LL_EXTI_IsActiveFallingFlag_0_31(line)
+#define LL_EXTI_IsActiveFlag_0_31(line) \
+	(EXTI_IS_ACTIVE_RISING_FLAG_0_31(line) || EXTI_IS_ACTIVE_FALLING_FLAG_0_31(line))
+
+#define EXTI_CLEAR_RISING_FLAG_0_31(line)  LL_EXTI_ClearRisingFlag_0_31(line)
+#define EXTI_CLEAR_FALLING_FLAG_0_31(line) LL_EXTI_ClearFallingFlag_0_31(line)
+#define LL_EXTI_ClearFlag_0_31(line) \
+	{ EXTI_CLEAR_RISING_FLAG_0_31(line); EXTI_CLEAR_FALLING_FLAG_0_31(line); }
```

Comment:
I'm not sure if this construct is standard-compliant or a GCC extension. Please use the construct Zephyr's macros seem to use (+ formatting):
```suggestion
#define LL_EXTI_ClearFlag_0_31(line)			\
	do {						\
		EXTI_CLEAR_RISING_FLAG_0_31(line);	\
		EXTI_CLEAR_FALLING_FLAG_0_31(line);	\
	} while(0)
```

Also if `EXTI_IS_ACTIVE_RISING_FLAG_0_31` et al. are only used as part of the macros, avoid them and use LL functions directly.

Reply:
Curly brackets are C-standard feature and defining scope (of visibility): https://en.cppreference.com/w/c/language/scope Thus it shall work with compilers other than gcc.

Reply:
After checking, it does seem to be fully standard; however, the `;` in `LL_EXTI_ClearFlag_0_31();` (i.e., when macro is used) becomes an empty statement. Most compilers don't seem to care about this though (clang requires `-Weverything`; GCC won't report even with `-Wall -Wextra -Wpedantic`, and neither will MSVC with `/Wall`) so I guess it's fine...
However, please do apply the formatting change.

Reply:
applied in https://github.com/zephyrproject-rtos/zephyr/commit/65f264ef9127701c7e85da8442ae169344879742